### DISCUSSION TITLE
ON TIMER does not trigger while waiting for INPUT$()

### DISCUSSION THREAD

**Discussion by Elvagor** [robhagemans/pcbasic#217] [Q&A]
As per the subject. Just wondering if this is expected/wanted.
Thanks!

**Comment by robhagemans** [comment]
I just checked with this test code:

    10 ON TIMER(1) GOSUB 1000
    15 TIMER ON
    20 A$=INPUT$(1): GOTO 20
    1000 PRINT TIMER: RETURN

Both GW-BASIC and PC-BASIC show the same behavior: the timer routine is not executed while waiting for input but instead is run once every time a key is hit. This makes sense, as the interpreter can't execute event loops while being stuck in a given function (in this case, `INPUT$`).

So yes, it's expected behaviour.

**Comment by Elvagor** [comment]
Thanks Rob!
